src/shared/sleep-config.c

```diff
@@ -164,10 +164,12 @@ int can_sleep_disk(char **types) {
 }
 
 #define HIBERNATION_SWAP_THRESHOLD 0.98
+#define MAX_SWAP_ENTRIES 32
 
 int find_hibernate_location(char **device, char **type, size_t *size, size_t *used) {
         _cleanup_fclose_ FILE *f;
-        unsigned i;
+        SwapEntry *swaps[MAX_SWAP_ENTRIES];
```

Comment:
Please declare this as as an `SwapEntry *swaps = NULL` and use GREEDY_REALLOC0 to resize the array on demand. No need for the arbitrary limit of 32 entries.